zebra/zebra_dplane.c

```diff
@@ -5617,16 +5625,15 @@ int dplane_provider_register(const char *name,
 			 "provider-%u", p->dp_id);
 
 	/* Insert into list ordered by priority */
-	TAILQ_FOREACH(last, &zdplane_info.dg_providers_q, dp_prov_link) {
+	frr_each (dplane_prov_list, &zdplane_info.dg_providers, last) {
 		if (last->dp_priority > p->dp_priority)
 			break;
 	}
 
 	if (last)
-		TAILQ_INSERT_BEFORE(last, p, dp_prov_link);
+		dplane_prov_list_add_after(&zdplane_info.dg_providers, last, p);
```

Comment:
This is OK, but I'd like to leave a comment for other reviewers:
`_add_after` actually inserts item `p` before `last`. The [documentation](http://docs.frrouting.org/projects/dev-guide/en/latest/lists.html#c.Z_add_after) suggests flipping the parameters, but I think renaming to `_add_before` would be more appropriated (and would avoid changing function prototype / breaking existing things)...

Reply:
<curse word> - sorry, I didn't realize that. I'll fix it - I do want the correct ordering...

Reply:
Just to clear up: there is nothing to fix, the issue (IMO) is actually in the library function naming.